app/src/main/kotlin/com/wire/android/notification/MessageNotificationManager.kt

```diff
@@ -26,41 +30,40 @@ import javax.inject.Singleton
 
     private val notificationManager = NotificationManagerCompat.from(context)
+    private val oldNotificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
 
-    private var prevNotificationsData: List<LocalNotificationConversation> = listOf()
+    fun handleNotification(newNotifications: List<LocalNotificationConversation>, userId: QualifiedID?) {
 
-    init {
-        appLogger.i("$TAG: initialized")
-    }
-
-    fun handleNotification(
-        newData: List<LocalNotificationConversation>,
-        userId: QualifiedID?
-    ) {
-        val oldData = prevNotificationsData
-        val oldConversationIds = oldData.map { it.id }
-        val newConversationIds = newData.map { it.id }
-
-        val conversationIdsToRemove = oldConversationIds.filter { !newConversationIds.contains(it) }
-        val conversationsToAdd = newData
-            .filter { conversation ->
-                val oldConversation = oldData.firstOrNull { it.id == conversation.id }
-                oldConversation == null || oldConversation != conversation
-            }
-            .map { it.intoNotificationConversation() }
+        if (newNotifications.isEmpty()) return
 
+        val activeNotifications: Array<StatusBarNotification> = oldNotificationManager.activeNotifications ?: arrayOf()
         val userIdString = userId?.toString()
 
         createNotificationChannel()
-        conversationsToAdd.forEach { showConversationNotification(it, userIdString) }
-        conversationIdsToRemove.forEach { hideNotification(it) }
-        showSummaryIfNeeded(oldData, newData, userIdString)
-
-        appLogger.i(
-            "$TAG: handled notifications: oldDataSize ${oldData.size}; newDataSize ${newData.size}; " +
-                "${conversationsToAdd.size} notifications were added; ${conversationIdsToRemove.size} notifications were removed. "
-        )
-        prevNotificationsData = newData
+        showSummaryIfNeeded(activeNotifications)
+        newNotifications.forEach {
+            showConversationNotification(it.intoNotificationConversation(), userIdString, activeNotifications)
+        }
+
+        appLogger.i("$TAG: handled notifications: newNotifications size ${newNotifications.size}; ")
+    }
+
+    fun hideNotification(conversationsId: ConversationId) {
+        notificationManager.cancel(getConversationNotificationId(conversationsId))
+
+        val notificationsLeft = oldNotificationManager.activeNotifications
+            ?.filter { it.groupKey.endsWith(NotificationConstants.MESSAGE_GROUP_KEY) }
+            ?.size ?: 0
+
+        if (notificationsLeft <= 1) {
+            // only SummaryNotification left
+            notificationManager.cancel(NotificationConstants.MESSAGE_SUMMARY_ID)
+        }
+    }
+
+    fun hideAllNotifications() {
+        // removing groupSummary removes all the notifications in a group
+        notificationManager.cancel(NotificationConstants.MESSAGE_SUMMARY_ID)
```

Comment:
Is this `MESSAGE_SUMMARY_ID` grouping ALL kind of notifications or only text related? Just asking out of ignorance here.

Reply:
it groups everything except the calls, calls are in the separate channel